packages/nitro-rpc-client/src/rpc-client.ts

```diff
@@ -57,17 +59,47 @@ export class NitroRpcClient implements RpcClientApi {
     return getAndValidateResult(res, "receive_voucher");
   }
 
-  public async WaitForObjective(objectiveId: string): Promise<void> {
-    return new Promise((resolve) => {
+  public async WaitForLedgerChannelToHaveStatus(
+    channelId: string,
+    status: ChannelStatus
+  ): Promise<void> {
+    const promise = new Promise<void>((resolve) => {
+      this.transport.Notifications.on(
+        "ledger_channel_updated",
+        (payload: LedgerChannelUpdatedNotification["params"]["payload"]) => {
+          if (payload.ID === channelId) {
+            this.GetLedgerChannel(channelId).then((l) => {
+              if (l.Status == status) resolve();
+            });
+          }
+        }
+      );
+    });
+    const ledger = await this.GetLedgerChannel(channelId);
```

Comment:
Is there need for both this `GetLedgerChannel` call and the same one above nested inside the `Notifications` handler?

Reply:
What I am trying to do here is make it impossible for there to be a race condition. So the approach is, 
Step 1: register a handler in case the event has not yet happened (if it already happened, this will never fire)
Step 2: make a static query in case the event has already happened. 

Which should result in the promise resolving whenever the event fires / was fired. 

Does that make sense?